internal/gcsx/multi_range_downloader_wrapper.go

```diff
@@ -229,27 +230,30 @@ func (mrdWrapper *MultiRangeDownloaderWrapper) Read(ctx context.Context, buf []b
 		}()
 
 		if e != nil && e != io.EOF {
-			e = fmt.Errorf("Error in Add Call: %w", e)
+			e = fmt.Errorf("error in Add call: %w", e)
 		}
 	})
 
-	select {
-	case <-time.After(timeout):
-		err = fmt.Errorf("Timeout")
-	case <-ctx.Done():
-		err = fmt.Errorf("Context Cancelled: %w", ctx.Err())
-	case res := <-done:
+	if !mrdWrapper.config.FileSystem.IgnoreInterrupts {
+		select {
+		case <-time.After(timeout):
```

Comment:
we can remove the timeout since its currently set to a very higher value @abhishek10004